examples/auth-rust-demo/tests/src/tests.rs

```diff
@@ -0,0 +1,232 @@
+
+#[derive(Clone)]
+struct CKbAuth {
+    privkey: Privkey,
+}
+impl CKbAuth {
+    fn new() -> Self {
+        Self {
+            privkey: Generator::random_privkey(),
+        }
+    }
+
+    fn get_pubkey_hash(&self) -> [u8; 20] {
+        let pub_key = self.privkey.pubkey().expect("pubkey").serialize();
+        let pub_hash = blake2b_256(pub_key.as_slice());
+        pub_hash[0..20].try_into().unwrap()
+    }
+
+    fn get_auth_args(&self, entry_category_type: &EntryCategoryType) -> Bytes {
+        struct EntryType {
+            _code_hash: [u8; 20],
+            _hash_type: u8,
+            _entry_category: u8,
+        }
+        let mut bytes = BytesMut::with_capacity(size_of::<CkbAuthType>() + size_of::<EntryType>());
+
+        let ckb_auth_type = CkbAuthType {
+            algorithm_id: AuthAlgorithmIdType::Ckb,
+            pubkey_hash: self.get_pubkey_hash(),
+        };
+        bytes.put_u8(ckb_auth_type.algorithm_id.into());
+        bytes.put(Bytes::from(ckb_auth_type.pubkey_hash.to_vec()));
+
+        bytes.put(CellOutput::calc_data_hash(&Loader::default().load_auth()).as_bytes());
+        bytes.put_u8(ScriptHashType::Data1.into());
+        bytes.put_u8(entry_category_type.clone() as u8);
+
+        bytes.freeze()
+    }
+
+    fn sign(&self, msg: &[u8; 32]) -> Bytes {
+        // convert message
+        let msg = H256::from(msg.clone());
+
+        let sig = self
+            .privkey
+            .sign_recoverable(&msg)
+            .expect("sign")
+            .serialize();
+        Bytes::from(sig)
+    }
+    fn sign_size(&self) -> usize {
+        // The fixed length of ckb signature is 65
+        65
+    }
+}
+
+fn gen_tx(ctx: &mut Context, grouped_args: Vec<(Bytes, usize)>) -> TransactionView {
+    let mut rng = thread_rng();
+    let loader = Loader::default();
+
+    let auth_demo_outpoint = ctx.deploy_cell(loader.load_demo());
+    let sighash_dl_out_point = ctx.deploy_cell(loader.load_auth());
+    let secp256k1_data_out_point = ctx.deploy_cell(loader.load_secp256k1_data());
+
+    // setup default tx builder
+    let dummy_capacity = Capacity::shannons(42);
+    let mut tx_builder = TransactionBuilder::default()
+        .cell_dep(
+            CellDep::new_builder()
+                .out_point(sighash_dl_out_point)
+                .dep_type(DepType::Code.into())
+                .build(),
+        )
+        .cell_dep(
+            CellDep::new_builder()
+                .out_point(secp256k1_data_out_point)
+                .dep_type(DepType::Code.into())
+                .build(),
+        )
+        .output(
+            CellOutput::new_builder()
+                .capacity(dummy_capacity.pack())
+                .build(),
+        )
+        .output_data(Bytes::new().pack());
+
+    for (args, inputs_size) in grouped_args {
+        // setup dummy input unlock script
+        for _ in 0..inputs_size {
+            let script = ctx
+                .build_script_with_hash_type(
```

Comment:
suggest to use `build_script` which will use `type` as the script hash type:  
https://github.com/nervosnetwork/capsule/blob/5e49142f25a17639122c2022d5437a5b258ec81f/crates/testtool/src/context.rs#L207

Reply:
`ScriptHashType::Data2` is used here, build_script defaults to `Type`

Reply:
If I understand correctly, ckb-auth-rs supports `Type` also, and `Data2` is only available in ckb2023 hardfork, which is not activated in mainnet and testnet, may cause misunderstandings for developers.